modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -310,7 +333,267 @@ void run(ClusterState state) {
+                    // target downsampling index exists and is not a downsampling index (name clash?)
+                    // we fail now but perhaps we should just randomise the name?
+                    String previousError = errorStore.getError(indexName);
+
+                    errorStore.recordError(indexName, new ResourceAlreadyExistsException(targetDownsampleIndexMeta.getIndex().getName()));
+                    // To avoid spamming our logs, we only want to log the error once.
+                    if (previousError == null || previousError.equals(errorStore.getError(indexName)) == false) {
+                        logger.error(
+                            "Data stream lifecycle service is unable to downsample backing index [{}] for data stream [{}] and "
+                                + "donwsampling round [{}] because the target downsample index [{}] already exists",
+                            indexName,
+                            dataStream.getName(),
+                            round,
+                            downsampleIndexName
+                        );
+                    }
+                    break;
+                } else if (downsampleStatus.equals(STARTED)) {
+                    // we'll wait for this round to complete
+                    // TODO add support for cancelling a current in-progress operation if another, later, round matches
+                    logger.trace(
+                        "Data stream lifecycle service waits for index [{}] to be downsampled. Current status is [{}] and the "
+                            + "downsample index name is [{}]",
+                        indexName,
+                        STARTED,
+                        downsampleIndexName
+                    );
+                    affectedIndices.add(index);
+                    break;
+                } else if (downsampleStatus.equals(SUCCESS)
+                    && dataStream.getIndices().contains(targetDownsampleIndexMeta.getIndex()) == false) {
+                        // at this point the source index is part of the data stream and the downsample index is complete but not
+                        // part of the data stream. we need to replace the source index with the downsample index in the data stream
+                        // and delete the source index.
+                        affectedIndices.add(index);
+                        replaceBackingIndexWithDownsampleIndexOnce(dataStream, indexName, downsampleIndexName);
+                        break;
+                    }
+            } else {
+                if (round.equals(lastRound)) {
+                    // no maintenance needed for previously started downsampling actions and we are on the last matching round so it's time
+                    // to kick off downsampling
+                    affectedIndices.add(index);
+                    DownsampleAction.Request request = new DownsampleAction.Request(indexName, downsampleIndexName, null, round.config());
```

Comment:
Out of curiosity, is 1 day (the default wait timeout) enough for this? What would happen if the downsampling always took longer than a day? Should we include a way to increase or set this?

Reply:
Added a disruption IT https://github.com/elastic/elasticsearch/pull/98609/files#diff-f7df158addbd7b19e481e80affffaad32f58cf4d5f2a1b400e9d7b584bf7a6daR40